src/assets/src/components/common.tsx

```diff
@@ -350,21 +389,28 @@ export const ShowRemainingField: React.FC<ShowRemainingFieldProps> = (props) =>
         setEditorValue(props.text);
     }
 
+    const statelessProps = {
+        id: props.id,
+        onSubmit: submit,
+        value: editorValue,
+        onChangeValue: setEditorValue,
+        placeholder: props.placeholder,
+        validateLength: props.validateLength,
+        maxLength: props.maxLength,
+        allowBlank: props.allowBlank === undefined ? true : props.allowBlank,
+        buttonType: props.buttonType,
+        disabled: props.disabled,
+    };
+
+    let statelessComponent = null;
+    if (props.fieldType === EditToggleFieldType.textArea) {
+        statelessComponent = <StatelessTextAreaForm {...statelessProps}>{props.children}</StatelessTextAreaForm>;
+    } else if (props.fieldType === EditToggleFieldType.inputGroup) {
+        statelessComponent = <StatelessInputGroupForm {...statelessProps}>{props.children}</StatelessInputGroupForm>;
+    }
```

Comment:
I imagine there is a better way to do this, but I haven't found it yet. Still, this seems preferable over duplicating the editing `useState` hooks and conditional "Edit" button in `EditToggleField` and `ShowRemainingField`.

Reply:
Hmm, what do you think of this?:
```typescript
interface EditToggleFieldProps {
    fieldComponent: React.FC<StatelessValidatedInputFormProps>;
    text: string;
    placeholder: string;
    disabled: boolean;
    buttonType: BootstrapButtonTypes;
    id: string;
    onSubmit: (value: string) => void;
    initialState: boolean;
    validateLength: boolean;
    maxLength?: number;
    allowBlank?: boolean;
}

export const EditToggleField: React.FC<EditToggleFieldProps> = (props) => {
    const [editing, setEditing] = useState(props.initialState);
    const [editorValue, setEditorValue] = useState(props.text);
    const submit = (value: string) => {
        props.onSubmit(value);
        setEditing(false);
    }
    const enableEditMode = () => {
        setEditing(true);
        setEditorValue(props.text);
    }

    const statelessProps: StatelessValidatedInputFormProps = {
        id: props.id,
        onSubmit: submit,
        value: editorValue,
        onChangeValue: setEditorValue,
        placeholder: props.placeholder,
        validateLength: props.validateLength,
        maxLength: props.maxLength,
        allowBlank: props.allowBlank ?? true,
        buttonType: props.buttonType,
        disabled: props.disabled,
    };
    const statelessComponent = <props.fieldComponent {...statelessProps} />
    return (editing && !props.disabled)
        ? (statelessComponent)
        : (
            <div className="input-group">
                <span>{props.text}</span>
                <button onClick={enableEditMode} type="button" className="btn btn-sm">
                    <FontAwesomeIcon icon={faPencilAlt} /> Edit
                </button>
            </div>
        );
}
```
This removes the enum in favor of passing in the component itself, and TS will validate that the type meets our requirements for props (`StatelessValidatedInputFormProps`)

Reply:
Good idea! I like it, I'll make the change.

Reply:
Done!